src/core_plugins/kibana/public/dashboard/dashboard_constants.js

```diff
@@ -2,7 +2,10 @@
 export const DashboardConstants = {
   ADD_VISUALIZATION_TO_DASHBOARD_MODE_PARAM: 'addToDashboard',
   NEW_VISUALIZATION_ID_PARAM: 'addVisualization',
-  LANDING_PAGE_PATH: '/dashboard',
-  CREATE_NEW_DASHBOARD_URL: '/dashboard/create',
-  EXISTING_DASHBOARD_URL: '/dashboard/{{id}}'
+  LANDING_PAGE_PATH: '/dashboard/list',
+  // NOTE, the following two urls have to have LANDING_PAGE_PATH as their sub url for
+  // the chrome nav links to work correctly (eg. so navigating away from an opened dashboard, and back to
+  // the dashboard app, will reopen the dashboard, not show the landing page.
```

Comment:
I believe this is because of the way the chrome app link url is determined (See my comment below).  Didn't seem to work when the app link url was `dashboard\list` but the other links were `dashboard\create` and `dashboard\edit..`.   It wasn't because of the id thing (though potentially could be an issue if a old dashboard was titled `edit` or `create` and had that as the id).
 
As per comment below, I didn't have time to track down the logic that figured out the chrome app link urls, so this is my theory since it started working as soon as the landing page/app link url was a substring of the other urls.